front/lib/swr/swr.ts

```diff
@@ -13,19 +14,57 @@ export function useSWRWithDefaults<TKey extends Key, TData>(
   fetcher: Fetcher<TData, TKey> | null,
   config?: SWRConfiguration & { disabled?: boolean }
 ) {
-  const { mutate: globalMutate } = useSWRConfig();
+  const { mutate: globalMutate, cache } = useSWRConfig();
 
   const mergedConfig = { ...DEFAULT_SWR_CONFIG, ...config };
   const disabled = !!mergedConfig.disabled;
 
   const result = useSWR(disabled ? null : key, fetcher, mergedConfig);
 
+  const mutateKeysWithSameUrl = useCallback(
+    (key: TKey) => {
+      // If the key looks like an url, we need to remove the query params
+      // to make sure we don't cache different pages together
+      // Naive way to detect url by checking for '/'
+      if (typeof key === "string" && key.includes("/")) {
+        const keyWithoutQueryParams = key.split("?")[0];
+
+        // Cycle through all the keys in the cache that start with the same url
+        // and mutate them too
+        for (const k of cache.keys()) {
+          if (
+            k !== key &&
+            typeof k === "string" &&
+            k.startsWith(keyWithoutQueryParams)
```

Comment:
This approach seems too broad. Suppose you have the following endpoints:
- `/api/vaults/[vId]`
- `/api/vaults/[vId]/data_source_views/[dsvId]`

If we perform a mutation on the vault, this code will likely match any endpoint that starts with `/api/vaults/[vId]`, including the data_source_views endpoint, potentially leading to unintended data mutations. Am I missing something?

If we use the `URL` module, we can actually do a strict equal on the route.

Reply:
You are correct, fixing.